e2e/upgrade_test.go

```diff
@@ -0,0 +1,97 @@
+package e2e
```

Comment:
Let's move the entire `e2e` directory current at the root of the agent repository to `testing/e2e` so that all of the test code and tooling is located together under the `testing` directory.